### PR TITLE
Filter Errors

### PR DESCRIPTION
**Pull request by Rythmag**
1. FilterErrors class is used to filter out repeated errors and logs which are a part of call stack
2. Bulk store method used to store found errors. Previous storeErrors method removed.

### CODE REVIEW

File: src/main/java/com/google/error_analyzer/backend/FilterErrors.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            String logText = (String) hit.getSourceAsMap().get(LogFields.LOG_TEXT);
+            if (filterForError(logText)) {
+                Document document = new Document(hit.getId(), hit.getSourceAsString());
+                documentList.add(document);
+            }
+        }
+        return documentList.build();
+    }
+
+    private Boolean filterForError(String logText) {
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/FilterErrors.java:44]: Seems more like keepTheError.

File: src/main/java/com/google/error_analyzer/backend/LogDao.java [hunk 2]
```diff
@@ -103,10 +104,12 @@ public long getDocumentCount (String index) throws IOException {
     //Returns name of the new index 
     @Override 
     public String findAndStoreErrors(String fileName) throws IOException {
-        SearchHits hits = findErrors(fileName);
+        ImmutableList < Document > errorHits = findErrors(fileName);
         String errorFileName = LogDaoHelper.getErrorIndexName(fileName);
-        storeErrors(errorFileName, hits);
-        logger.info("Errors stored in index ".concat(errorFileName));
+        Integer size = errorHits.size();
+        logger.info("storing ".concat(size.toString())
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/LogDao.java:110]: Use string.format()

File: src/main/java/com/google/error_analyzer/data/Document.java [hunk 3]
```diff
@@ -47,4 +48,23 @@ public String getID() {
         return id;
     }
 
+    public boolean equals(Document other) {
+        return (other.getID().equals(this.id) && other.getJsonString().equals(this.jsonString));
+    }
+
+    public static Boolean compareDocumentList(List < Document > expected, List < Document > actual) {
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/data/Document.java:55]: Where is this getting used?

Comment (Rythmag) [reply src/main/java/com/google/error_analyzer/data/Document.java:55]: FilterErrorsTest

Comment (Rythmag) [reply src/main/java/com/google/error_analyzer/data/Document.java]: Removed the function and changed Document.equals() function

File: src/test/java/com/google/error_analyzer/backend/FilterErrorsTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import org.mockito.*;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+import org.springframework.test.util.ReflectionTestUtils;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doAnswer;
```
Comment (nishantnahata) [inline src/test/java/com/google/error_analyzer/backend/FilterErrorsTest.java:36]: Unused import.
Please remove any unused imports from the code

File: src/main/java/com/google/error_analyzer/backend/LogDao.java [hunk 5]
```diff
@@ -103,10 +104,11 @@ public long getDocumentCount (String index) throws IOException {
     //Returns name of the new index 
     @Override 
     public String findAndStoreErrors(String fileName) throws IOException {
-        SearchHits hits = findErrors(fileName);
+        ImmutableList < Document > errorHits = findErrors(fileName);
         String errorFileName = LogDaoHelper.getErrorIndexName(fileName);
```
Comment (nishantnahata) [inline src/main/java/com/google/error_analyzer/backend/LogDao.java:108]: @asthaaaa please review this part.

Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/LogDao.java:108]: reviewed

File: src/main/java/com/google/error_analyzer/backend/FilterErrors.java [hunk 6]
```diff
@@ -42,12 +43,13 @@
...
 
     //return false if error is repeated or part of stack
-    private Boolean shouldIncludeError(String logText) {
+    private Boolean shouldIncludeError(String logText, Set < String > errorSet) {
         if (StackTraceFormat.matchesFormat(logText)) {
             return false;
         }
         logText = logText.replaceAll("[^a-zA-Z]", " ");
         if (errorSet.contains(logText)) {
+            System.out.println("Repeated error");
```
Comment (nishantnahata) [inline src/main/java/com/google/error_analyzer/backend/FilterErrors.java:52]: logger?

File: src/main/java/com/google/error_analyzer/backend/FilterErrors.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            if (shouldIncludeError(logText, errorSet)) {
+                Document document = new Document(hit.getId(), hit.getSourceAsString());
+                documentList.add(document);
+            }
+        }
+        return documentList.build();
+    }
+
+    //return false if error is repeated or part of stack
+    private Boolean shouldIncludeError(String logText, Set < String > errorSet) {
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/FilterErrors.java:49]: Optional:

Name : containsError

Comment (Rythmag) [reply src/main/java/com/google/error_analyzer/backend/FilterErrors.java:49]: I dont think this name address the issue of repeated error

**Pull request merged**